user-posts/pages/index.vue

```diff
@@ -46,70 +49,105 @@ if (totalData.value) {
 
 const totalPages = Math.ceil(totalPosts.value / limit.value);
 
-console.log(totalPosts.value);
-
 const goToPage = (page: number): void => {
   if (page >= 1 && page <= totalPages) {
     currentPage.value = page;
   }
 };
+
+function handleCreate() {
+  isModalVisible.value = true;
+  isUpdate.value = false;
+}
+
+const handleFormSubmit = (formData: Post) => {
+  if (isUpdate.value) {
+    const id = formData?.id;
+
+    if (id) {
+      const index = posts.value.findIndex((post) => post.id === id);
+
+      if (index !== -1) {
+        posts.value[index] = { ...posts.value[index], ...formData };
+      } else {
+        console.error("Post not found with id:", id);
+      }
+    } else {
+      console.error("Invalid post ID:", id);
+    }
+  } else {
+    const newObj = { ...formData };
+    fetch("https://jsonplaceholder.typicode.com/posts", {
+      method: "POST",
+      body: JSON.stringify({
+        title: newObj.title,
+        body: newObj.body,
+        userId: newObj.userId,
+      }),
+      headers: {
+        "Content-type": "application/json; charset=UTF-8",
+      },
+    })
+      .then((response) => response.json())
+      .then((json) => posts.value.push(json));
+  }
+};
+
+const handleEdit = (post: Post) => {
+  isModalVisible.value = true;
+  isUpdate.value = true;
+  selectedPost.value = { ...post };
+};
+
```

Comment:
Refactor this ... use https://ui.nuxt.com/components/notification to show toast for error or successful CRUD